packaging/Microsoft.NET.Sdk.WindowsDesktop/targets/Microsoft.NET.Sdk.WindowsDesktop.targets

```diff
@@ -27,11 +27,30 @@
 
   <ItemGroup Condition=" ('$(EnableDefaultItems)' == 'true') And ('$(UseWPF)' == 'true') And ('$(_TargetFrameworkVersionValue)' != '$(_UndefinedTargetFrameworkVersion)') And 
                          ('$(_TargetFrameworkVersionValue)' >= '$(_WindowsDesktopSdkTargetFrameworkVersionFloor)')">
-    
-    <!-- In the WindowsDesktop .props, we globbed all .xaml files as Page items.  If any of those files are included
-         as Resource, Content, or None items, then remove them from the Page items. -->
-    <Page Remove="@(Resource);@(Content);@(None)"
+
+    <!--
+         In the WindowsDesktop .props, we globbed all .xaml files as Page items.  If any of those files are included
+         as Resource, Content then remove them from the Page items.
+    -->
+    <Page Remove="@(Resource);@(Content)"
           Condition="'$(EnableDefaultPageItems)' != 'false'" />
+
+    <!--
+        If ApplicationDefinition must also be removed from Page as PresentationBuildTasks will include the
+        generated code for the ApplicationDefinition twice otherwise.  We cannot do this in all cases in the
+        props file as ApplicationDefinition won't be available at that point in the evaluation when
+        EnableDefaultApplicationDefinition is false.  We unconditionally remove it here as ApplicationDefinition
+        is considered a special singleton Item that WPF ensures is only where it needs to be.
+    -->
+    <Page Remove="@(ApplicationDefinition)" />
+    
+    <!--
+         Only remove None from Page in the event that we disable both the default ApplicationDefinition and the DefaultPageItems.
```

Comment:
I think that this comment doesn't accurately reflect the code (though it's hard to tell with all the negatives).  I think it would be correct to replace "disable" with "enable".

Also I'm not following why EnableDefaultApplicationDefinition should affect whether `None` is removed from `Page`.

Reply:
If you set EnableDefaultApplicationDefinition to false, then we don't remove anything from none here: https://github.com/dotnet/wpf/blob/f0c10e1364f1230b053ce6f2c9832817f88a9758/packaging/Microsoft.NET.Sdk.WindowsDesktop/targets/Microsoft.NET.Sdk.WindowsDesktop.props#L64

This means None will contain page items like MainWindow.xaml.  If you then remove None from Page, you'll remove items that should be there.

The change to none was intentional (see the comment above) since VS couldn't handle more specific globbing fixups to None.  In order to preserve that and fix the issue from this bug, we have to condition removing None from Page.

Reply:
Updated comments to better explain the conditions.